### DISCUSSION TITLE
How to get sortedData using useTable

### DISCUSSION THREAD

**Discussion by candidia** [TanStack/table#2154] [General] [answered]
Have found a few answers on getting the data as it is displayed using the older versions of this library like https://stackoverflow.com/questions/48120858/access-filtered-data-in-reacttable but I can't figure out how to in the new version. 

Tried accessing `state` but I only get the sortby column and such.

Looking at the instance props, I can't find one that will show me the data as it is displayed https://react-table.js.org/api/useTable#instance-properties

**Comment by candidia** [comment, marked as answer]
Just kidding, `rows` is all I need

**Comment by mnai01** [comment]
Extract 'rows' from the destructured values and that will give you all the row data that has been filtered

```js
const {
    getTableProps,
    headerGroups,
    prepareRow,
    page,
    gotoPage,
    setPageSize,
    rows, // <-------------
    state: { pageIndex, pageSize, filters }
  } = useTable(
    { data, columns, autoResetHiddenColumns: false },
    useFilters,
    useGlobalFilter,
    useSortBy,
    usePagination,
    useRowSelect
  );
  ```
